### PR TITLE
Update package.py for HDF5 1.12.0 release

### PR DESCRIPTION
**Pull request by lkurz**
Update HDF5 spec for latest release 1.12.0.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -36,8 +38,8 @@ class Hdf5(AutotoolsPackage):
     version('1.8.19', sha256='a4335849f19fae88c264fd0df046bc321a78c536b2548fc508627a790564dc38')
     version('1.8.18', sha256='cdb195ad8d9e6782acf24b2488061289f615628c2ccda8457b0a0c3fb7a8a063')
     version('1.8.17', sha256='d9cda297ee76ade9881c4208987939250d397bae6252d0ccb66fa7d24d67e263')
-    version('1.8.16', sha256='ed17178abd9928a7237f30370189ba767b9e39e0db45917c2ac4665eb9cb4771')
     version('1.8.15', sha256='4e963216b7d32469596bc1321a8c3f6e0c278dcbbdb7be6414c63c081b34c275')
+    version('1.8.16', sha256='ed17178abd9928a7237f30370189ba767b9e39e0db45917c2ac4665eb9cb4771')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Why this change?

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 2]
```diff
@@ -230,7 +232,7 @@ def configure_args(self):
         else:
             extra_args.append('--without-szlib')
 
-        if self.spec.satisfies('@1.10:'):
+        if self.spec.satisfies('@1.10:1.12.0'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Ah, I didn't realize we were already adding this, you can undo this change. Sorry for the confusion.

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 1 above)
Comment (lkurz) [reply var/spack/repos/builtin/packages/hdf5/package.py]: This was an unintended reordering.  
I'll reverse the change.

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 2 above)
Comment (lkurz) [reply var/spack/repos/builtin/packages/hdf5/package.py]: No problem. I'll remove the change.

**Comment by roblatham00** [comment]
This commit looks really strange:  just one line adding a new version?

**Comment by adamjstewart** [comment]
@roblatham00 that's all it takes to add a new version of a package

**Comment by roblatham00** [comment]
right, i got that, but you yourself requested changes to the `configure_args` handling and there were other modifications that got dropped.  Are those changes going into a different pull request?

**Comment by adamjstewart** [comment]
Ah, see the resolved comments. I realized that we are already specifying `--enable-build-mode` for all versions so no change was necessary.

**Comment by ax3l** [comment]
@lkurz can we please add a HDF5 package variant for the HDF5 API compatibility? This new major release broke a lot of Spack package builds. #15502

Probably a good amount of the ones listed in `spack dependents hdf5`.

**Comment by lkurz** [comment]
Thanks for the report Axel.
I'll get the team working on it tonight.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ax3l** [comment]
P.S.: congrats on the new major release!

**Comment by adamjstewart** [comment]
@lkurz while you're at it, would you like to add yourself as a `maintainers`? That way we know who to contact if anyone is having problems with the HDF packages in Spack.
